signer/signer.go

```diff
@@ -0,0 +1,6 @@
+package signer
+
+type Signer interface {
+	Sign() interface{}   // can be a signature type (e.g. secp256k1, []byte, etc.)
```

Comment:
```suggestion
	Sign(messageDigest []byte) interface{}   // can be a signature type (e.g. secp256k1, []byte, etc.)
```

I believe the message digest generated by hashing original message should be required to generate a signature (see [FIPS 186-4: Digital Signature Standard](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf)).

Reply:
i missed that parameter :( thx